examples/ping_pong.js

```diff
@@ -1,24 +1,27 @@
-var frida = require('..');
+'use strict';
 
-var processName = process.argv[2];
+const frida = require('..');
 
-var script =
-"recv('poke', function onMessage(pokeMessage) {" +
-"  send('pokeBack');"                            +
-"});";
+const processName = process.argv[2];
 
-frida.attach(processName).then(function (session) {
-  return session.createScript(script);
-}).then(function (script) {
-  script.events.listen('message', function (message) {
+let script =
+`recv('poke', function onMessage(pokeMessage) {
+  send('pokeBack');
+});`;
+
+frida.attach(processName)
+.then(session => session.createScript(script))
+.then(script => {
+  script.events.listen('message', message => {
     console.log(message);
   });
-  script.load().then(function () {
-    console.log("script loaded");
-    script.postMessage({ "type": "poke" });
-  }).catch(function (err) {
-    console.error(err);
-  });
-}).catch(function (err) {
+
+  return script.load();
+})
+.then(() => {
+  console.log("script loaded");
+  script.postMessage({ "type": "poke" });
```

Comment:
This looks like a regression – the `script` variable is bound to the source-code and not the script that was returned by `session.createScript()`.